lib/tasks/server/express-server.js

```diff
@@ -213,6 +215,21 @@ class ExpressServerTask extends Task {
     let options = this.startOptions;
     options.httpServer = this.httpServer;
 
+    let liveReloadServer = new LiveReloadServer({
+      app: this.app,
+      ui: options.ui,
+      watcher: options.watcher,
+      analytics: options.analytics,
+      project: options.project,
+      httpServer: options.httpServer,
+      port: options.port,
+      prefix: options.liveReloadPrefix,
+    });
+    let middleware = liveReloadServer.getMiddleware(this.startOptions);
```

Comment:
we spoke briefly offline, this code-path may spawn a promise which may reject. That can occur if a custom live-reload port is used, but the port isn't free.

The way it is currently implemented, `getMiddleware` doesn't return a promise, so we can't receive the rejection.

One likely approach, is to change `getMiddleware` to `setupMiddleware`, which internally appends the middleware, and returns a promise, that can be part of the promise chain bellow. 

This should connect all the pieces, making both tests easier, and ensuring error handling continues to work correctly.